lib/rubocop/cop/rspec/metadata_style.rb

```diff
@@ -0,0 +1,197 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Use consistent metadata style.
+      #
+      # This cop does not support autocorrection in the case of
+      # `EnforcedStyle: hash` where the trailing metadata type is ambiguous.
+      # (e.g. `describe 'Something', :a, b`)
+      #
+      # @example EnforcedStyle: symbol (default)
+      #   # bad
+      #   describe 'Something', a: true
+      #
+      #   # good
+      #   describe 'Something', :a
+      #
+      # @example EnforcedStyle: hash
+      #   # bad
+      #   describe 'Something', :a
+      #
+      #   # good
+      #   describe 'Something', a: true
+      class MetadataStyle < Base # rubocop:disable Metrics/ClassLength
```

Comment:
The `Metrics/ClassLength` is spot on, though. This class is so long that I find it hard to read. But extracting logic into classes (e.g. `remove_pair(corrector, node)` → `RemovePair.new(corrector, node).call`) is quite hard.

Reply:
Perhaps the corrector code can be extract to a private AutoCorrector class but I'm happy for this to be an unrelated refactoring - when and if someone wants to undertake it